request.go

```diff
@@ -15,16 +16,43 @@ import (
 // MaxFilelist is the max number of files to return in a readdir batch.
 var MaxFilelist int64 = 100
 
+// CheckFileRequest contains the data for a file hashing request
+type CheckFileRequest struct {
+	HashAlgoList string
+	StartOffset  uint64
+	Length       uint64
+	BlockSize    uint32
+}
+
+// GetHashAlgoAsList returns the comma separataed HashAlgoList string as slice
+func (r CheckFileRequest) GetHashAlgoAsList() []string {
+	result := []string{}
+	for _, v := range strings.Split(r.HashAlgoList, ",") {
+		algo := strings.TrimSpace(v)
+		if len(algo) > 0 {
+			result = append(result, algo)
+		}
+	}
+	return result
+}
+
+// CheckFileResponse defines the response for file hashing requests
+type CheckFileResponse struct {
+	HashAlgoUsed string
+	Hash         []byte
+}
+
 // Request contains the data and state for the incoming service request.
 type Request struct {
 	// Get, Put, Setstat, Stat, Rename, Remove
 	// Rmdir, Mkdir, List, Readlink, Link, Symlink
-	Method   string
-	Filepath string
-	Flags    uint32
-	Attrs    []byte // convert to sub-struct
-	Target   string // for renames and sym-links
-	handle   string
+	Method    string
+	Filepath  string
+	Flags     uint32
+	Attrs     []byte // convert to sub-struct
+	Target    string // for renames and sym-links
+	handle    string
+	CheckFile CheckFileRequest // for CheckFile requests
```

Comment:
Not sure why this couldn’t be in its own separate “paragraph” and not cause a reflow of all the other fields’ whitespace.